recipes/magnum/all/conanfile.py

```diff
@@ -0,0 +1,723 @@
+        "gl": [True, False],
+        "mesh_tools": [True, False],
+        "primitives": [True, False],
+        "scene_graph": [True, False],
+        "shaders": [True, False],
+        "text": [True, False],
+        "texture_tools": [True, False],
+        "trade": [True, False],
+        "vk": [True, False],
+        
+        "android_application": [True, False],
+        "emscripten_application": [True, False],
+        "glfw_application": [True, False],
+        "glx_application": [True, False],
+        "sdl2_application": [True, False],
+        "xegl_application": [True, False],
+        "windowless_cgl_application": [True, False],
+        "windowless_egl_application": [True, False],
+        "windowless_glx_application": [True, False],
+        "windowless_ios_application": [True, False],
+        "windowless_wgl_application": [True, False],
+        "windowless_windows_egl_application": [True, False],
+
+        "cgl_context": [True, False],
+        "egl_context": [True, False],
+        "glx_context": [True, False],
+        "wgl_context": [True, False],
+
+        "gl_info": [True, False],
+        "al_info": [True, False],
+        "distance_field_converter": [True, False],
+        "font_converter": [True, False],
+        "image_converter": [True, False],
+        "scene_converter": [True, False],
+
+        # Options related to plugins
+        "any_audio_importer": [True, False],
+        "any_image_converter": [True, False],
+        "any_image_importer": [True, False],
+        "any_scene_converter": [True, False],
+        "any_scene_importer": [True, False],
+        "magnum_font": [True, False],
+        "magnum_font_converter": [True, False],
+        "obj_importer": [True, False],
+        "tga_importer": [True, False],
+        "tga_image_converter": [True, False],
+        "wav_audio_importer": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "shared_plugins": True,
+
+        "target_gl": "desktop_gl",
+        "target_headless": True,
+        "target_vk": True,
+
+        "audio": True,
+        "debug_tools": True,
+        "gl": True,
+        "mesh_tools": True,
+        "primitives": True,
+        "scene_graph": True,
+        "shaders": True,
+        "text": True,
+        "texture_tools": True,
+        "trade": True,
+        "vk": True,
+
+        "android_application": True,
+        "emscripten_application": True,
+        "glfw_application": True,
+        "glx_application": True,
+        "sdl2_application": True,
+        "xegl_application": True,
+        "windowless_cgl_application": True,
+        "windowless_egl_application": True,
+        "windowless_glx_application": True,
+        "windowless_ios_application": True,
+        "windowless_wgl_application": True,
+        "windowless_windows_egl_application": True,
+
+        "cgl_context": True,
+        "egl_context": True,
+        "glx_context": True,
+        "wgl_context": True,
```

Comment:
Do we care about some illegal option combinations/dependencies?
e.g. `windowless_cgl_application` is connected to `cgl_context`, likewise for `egl`, `glx`, `wgl`

Reply:
I see the project generates a `egl/glx/wgl` objects library, and it only generates the actual library when `xxx_context` is True. So, we can have `xxx_context=False` and the `xxx_application=True` and it will work.